### DISCUSSION TITLE
insert component between weekly DIV's

### DISCUSSION THREAD

**Discussion by trig79** [gpbl/react-day-picker#1323] [Support] [answered]
Hey 
I have the following calendar configured, it renders the timeslots when the user selects a calendar day.
![image](https://user-images.githubusercontent.com/65402006/145282524-1473a086-7503-4fea-9f48-fc58ec191855.png)

However, the client has requested to have the 'timeslots appear below the week they select like so:
![image](https://user-images.githubusercontent.com/65402006/145282804-0bf5ddd5-1cf5-4103-89b3-947b871179af.png)

Is there anything within this package that could support this idea? 

I was thinking perhaps react portal might be the way to go, but I'm struggling to get that to work at the moment.

any ideas welcome

thanks

**Comment by tjfred35** [comment, marked as answer]
You can do it with the v8 version like this:

```ts
  const [selected, setSelected] = React.useState<Date>();

  const CustomRow = (props: RowProps) => {
    return (
      <>
        <Row {...props} />
        {selected && props.weekNumber === getWeek(selected) && (
          <tr>
            <td colSpan={7}>
              <div
                style={{
                  padding: "16px 8px",
                  display: "flex",
                  justifyContent: "center",
                  margin: "4px 2px",
                  border: "1px solid blue",
                  borderRadius: "4px"
                }}
              >
                You selected me!
              </div>
            </td>
          </tr>
        )}
      </>
    );
  };

  return (
    <DayPicker
      mode="single"
      selected={selected}
      onSelect={setSelected}
      components={{
        Row: CustomRow
      }}
    />
  );
```

<img width="322" alt="Screen Shot 2021-12-10 at 4 50 05 PM" src="https://user-images.githubusercontent.com/2838338/145645673-02c93154-de08-4b12-a687-c21dedee3211.png">


https://codesandbox.io/s/sandpack-project-forked-9ltwx?file=/src/App.tsx:160-1020

**Reply by trig79** [reply]
Hey,  I had a play around with the new version and as you suggest it defo looks like this will work.
many thanks for taking the time to write such a comprehensive reply, I really do appreciate it.

**Reply by tjfred35** [reply]
No problem!
